integration-tests/src/bonds.rs

```diff
@@ -0,0 +1,124 @@
+#![cfg(test)]
+
+use crate::assert_balance;
+use crate::polkadot_test_net::*;
+
+use frame_support::{assert_noop, assert_ok};
+use orml_traits::MultiCurrency;
+use sp_runtime::BoundedVec;
+use xcm_emulator::TestExt;
+
+use hydradx_runtime::{AssetRegistry, Bonds, Currencies, Runtime, RuntimeOrigin};
+use primitives::constants::time::unix_time::MONTH;
+
+#[test]
+fn issue_bonds_should_work_when_issued_for_native_asset() {
+	Hydra::execute_with(|| {
+		// Arrange
+		let amount = 100 * UNITS;
+		let fee = <Runtime as pallet_bonds::Config>::ProtocolFee::get().mul_ceil(amount);
+		let amount_without_fee: Balance = amount.checked_sub(fee).unwrap();
+
+		let maturity = NOW + MONTH;
+
+		// Act
+		let bond_id = AssetRegistry::next_asset_id().unwrap();
+		assert_ok!(Bonds::issue(RuntimeOrigin::signed(ALICE.into()), HDX, amount, maturity));
+
+		// Assert
+		assert_eq!(Bonds::bonds(bond_id).unwrap(), (HDX, maturity));
+
+		let bond_asset_details = AssetRegistry::assets(bond_id).unwrap();
+
+		assert!(bond_asset_details.asset_type == pallet_asset_registry::AssetType::Bond);
```

Comment:
I would use for `assert_eq` for this to have a better failing message once it fails